### PR TITLE
FossGraph. Added list projects on CreateVulnerabilityView

### LINKED ISSUE #1991: FossGraph. Add list projects on CreateVulnerabilityView
Need add list projects on CreateVulnerabilityView

### PR DESCRIPTION
**Pull request by Cheshiriks**
* added list projects on CreateVulnerabilityView

С2332

Closes #1991

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt [hunk 1]
```diff
@@ -88,6 +131,43 @@ val createVulnerabilityView: VFC = VFC {
...
+                                                )
+                                            }
+                                            div {
+                                                className = ClassName("col-10 text-left align-self-center pl-0")
+                                                +url
+                                            }
+                                        }
+                                        div {
+                                            className = ClassName("col-1 align-self-right d-flex align-items-center justify-content-end")
+                                            button {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt:157]: why not buttinBuilder?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt [hunk 2]
```diff
@@ -68,8 +111,8 @@ val createVulnerabilityView: VFC = VFC {
                                         className = ClassName("input-group needs-validation")
                                         textarea {
                                             className = ClassName("form-control")
-                                            onChange = {
-                                                // FIXME: need to add method on backend
+                                            onChange = { event ->
+                                                setVulnerability(vulnerability.copy(description = event.target.value))
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt:115]: Better do it in functional style: 
```suggestion
                                                setVulnerability { it.copy(description = event.target.value) }
```

**Review by sanyavertolet — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt [hunk 3]
```diff
@@ -17,23 +22,61 @@ import react.dom.html.ReactHTML.h1
...
+            navigate(to = "/${FrontendRoutes.FOSS_GRAPH}")
+        }
+    }
+
+    displayModal(
+        deleteProjectWindowOpenness.isOpen(),
+        "Deletion of project",
+        "Are you sure you want to remove this project from the vulnerability?",
+        mediumTransparentModalStyle,
+        { deleteProjectWindowOpenness.closeWindow() },
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt:66]: you can use here `com.saveourtool.save.frontend.utils.WindowOpenness#closeWindowAction`,
if it expects some argument, there is the extension: `com.saveourtool.save.frontend.utils.UtilsKt#withUnusedArg`

**Review by nulls — CHANGES_REQUESTED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt [hunk 4]
```diff
@@ -17,23 +22,61 @@ import react.dom.html.ReactHTML.h1
...
+import kotlinx.serialization.json.Json
+
 val createVulnerabilityView: VFC = VFC {
     useBackground(Style.BLUE)
 
+    val navigate = useNavigate()
+
     val (vulnerability, setVulnerability) = useState(VulnerabilityDto.empty)
-    val (vulnerabilityProjects, setVulnerabilityProjects) = useState(mutableListOf<VulnerabilityProjectDto>())
+    val (vulnerabilityProjects) = useState(mutableListOf<VulnerabilityProjectDto>())
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/fossgraph/CreateVulnerabilityView.kt]: Fixed

**Review by nulls — APPROVED** [review]

**Pull request merged**